ws-client/src/transport.rs

```diff
@@ -227,7 +233,24 @@ impl<'a> WsTransportClientBuilder<'a> {
 					match self.mode {
 						Mode::Plain => TlsOrPlain::Plain(socket),
 						Mode::Tls => {
-							let connector = async_tls::TlsConnector::default();
+							let mut client_config = rustls::ClientConfig::default();
+							if let Some(path) = &self.custom_certificate {
+								// TODO: How to give use async for file io and still give add_pem_file() a std::io::BufRead?
+								let f = std::fs::File::open(&path)?;
+								let mut reader = io::BufReader::new(f);
+								let (added, failed) = client_config
```

Comment:
Do we want to expose or handle `added` and `failed` in any way differently?

For example, perhaps the contents of the `if let` should be in `custom_certificate()` which would then the result of `add_pem_file()` (plus any errors)

Reply:
No, I think it's fine to log it and return an error if the certificate could not be read from the Path.